api/src/main/java/org/pmiops/workbench/api/DataBrowserController.java

```diff
@@ -0,0 +1,292 @@
+            new Function<DbDomain, org.pmiops.workbench.model.DbDomain>() {
+                @Override
+                public org.pmiops.workbench.model.DbDomain apply(org.pmiops.workbench.cdr.model.DbDomain cdr) {
+                    return new org.pmiops.workbench.model.DbDomain()
+                            .domainId( cdr.getDomainId())
+                            .domainDisplay(cdr.getDomainDisplay())
+                            .domainDesc(cdr.getDomainDesc())
+                            .domainParent(cdr.getDomainParent())
+                            .domainRoute(cdr.getDomainRoute());
+
+
+                }
+            };
+
+
+
+    /**
+     * This method searches concepts
+     *
+     * @param conceptName
+     * @param standardConcept
+     * @param concept_code
+     * @param vocabulary_id
+     * @param domain_id
+     * @return
+     */
+    @Override
+    public ResponseEntity<ConceptListResponse> getConceptsSearch(
+            String conceptName,
+            String standardConcept,
+            String concept_code,
+            String vocabulary_id,
+            String domain_id) {
+
+
+        List<Concept> conceptList;
+
+        // If Concept name do search on name
+
+        if (conceptName != null) {
+            conceptList = conceptDao.findConceptLikeName(conceptName);
+        }
+        else {
+            conceptList = conceptDao.findConceptsOrderedByCount();
+        }
+
+        ConceptListResponse resp = new ConceptListResponse();
+        resp.setItems(conceptList.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+
+        return ResponseEntity.ok(resp);
+    }
+
+    /**
+     * This method gets concepts with maps to relationship in concept relationship table
+     *
+     * @param conceptId
+     * @param conceptNum
+     * @return
+     */
+    @Override
+    public ResponseEntity<ConceptListResponse> getConceptsMapsTo(
+            Long conceptId,
+            Long conceptNum) {
+
+
+        List<Concept> conceptList;
+
+        // If Concept num is 2 , children
+        if (conceptNum == 2 ) {
+            conceptList = conceptDao.findConceptsMapsToChildren(conceptId);
+        }
+        else {
+            conceptList = conceptDao.findConceptsMapsToParents(conceptId);
+        }
+
+        ConceptListResponse resp = new ConceptListResponse();
+        resp.setItems(conceptList.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+
+        return ResponseEntity.ok(resp);
+    }
+    /**
+     * This method searches concepts
+     *
+     * @param analysisId
+     * @param stratum1
+     * @param stratum2
+     * @param stratum3
+     * @param stratum4
+     * @param stratum5
+     * @return
+     */
+    @Override
+    public ResponseEntity<AnalysisResultListResponse> getAnalysisResults(
+            Long analysisId,
+            String stratum1,
+            String stratum2,
+            String stratum3,
+            String stratum4,
+            String stratum5) {
+
```

Comment:
Do you need the following params? stratum3, stratum4 or stratum5? You don't have a reference to them anywhere in this method.

Reply:
just for future analyses. analysis table has 5 stratum.

Reply:
I have not seen any mocks that propose slicing things by 5 dimensions within an analysis... originally we'd said there would just be a set of uni-dimensional breakdowns. 

I'd vote we add support for more dimensions at the point at which we need them. When it comes to APIs, less is more. :)